### PR TITLE
VS 396 clinvar grabs too many values

### PR DESCRIPTION
**Pull request by RoriCremer**
currently clinvar grabs all values in the variants section of the annotation json---this is wrong because nirvana includes clinvar values from all overlapping variants.

this change limits the clinvar values to the correct variant only
and tests the change as well

### CODE REVIEW

File: scripts/variantstore/wdl/extract/test_create_variant_annotation_table.py [hunk 1]
```diff
@@ -8,37 +8,105 @@
...
+            transcript_line=None)
+        self.maxDiff=None
+        self.assertEqual(actual.get('clinvar_classification'), ['likely pathogenic', 'pathogenic', 'conflicting data from submitters', 'guitar', 'kinney', 'sleater', 'solo'])
+        self.assertEqual(actual.get('clinvar_id'), ['RCV01', 'RCV02'])
+        self.assertEqual(actual.get('clinvar_phenotype'), ['blonde', 'brunette'])
+        self.assertEqual(actual.get('clinvar_last_updated'), '2020-03-02')
+
 
+    def test_clinvar_inclusion(self):
+        clinvar_swap = [{ 'id': 'RCV01', \
```
Comment (gbggrant) [inline scripts/variantstore/wdl/extract/test_create_variant_annotation_table.py:82]: Do you want to add a test case for a case where the id does NOT start with 'RCV'?

**Review by gbggrant — APPROVED** [review]

File: scripts/variantstore/wdl/extract/create_variant_annotation_table.py [hunk 2]
```diff
@@ -234,33 +235,37 @@ def make_annotated_json_row(row_position, row_ref, row_alt, variant_line, transc
...
+        # And then anything that did not match will get added at the end of the list (sorted alpha)
+        for value in significance_ordering:
+          if value in significance_values:
+            ordered_significance_values.append(value) # this adds the id to the end of the list
+        values_not_accounted_for = list(set(significance_values).difference(ordered_significance_values))
+        values_not_accounted_for.sort() # alphabetize this so it is deterministic
+        ordered_significance_values.extend(values_not_accounted_for) # add any values that aren't in significance_ordering to the end
+        row["clinvar_id"] = clinvar_ids # array
+        row["clinvar_classification"] = ordered_significance_values # special sorted array
+        updated_dates.sort(key=lambda date: datetime.strptime(date, "%Y-%m-%d")) # note: method is in-place, and returns None
```
Comment (mcovarr) [inline scripts/variantstore/wdl/extract/create_variant_annotation_table.py:266]: thank you for the comment, I forget that every time I've been away from Python for a while and it never fails to bite me 🙂

**Review by mcovarr — CHANGES_REQUESTED** [review]

File: scripts/variantstore/wdl/extract/test_create_variant_annotation_table.py (hunk 1 above)
Comment (RoriCremer) [reply scripts/variantstore/wdl/extract/test_create_variant_annotation_table.py:82]: Yes and I think that line 88 should handle that case

File: scripts/variantstore/wdl/extract/create_variant_annotation_table.py [hunk 3]
```diff
@@ -143,6 +143,7 @@ def get_gnomad_subpop(gnomad_obj):
     max_an = None
     max_af = None
     max_subpop = ""
+    ## TODO will there ever be unexpected values in gnomad_subpop (values not in gnomad_ordering) ?
```
[the remark this answers is not in the document]
Comment (RoriCremer) [reply scripts/variantstore/wdl/extract/create_variant_annotation_table.py:146]: yes I think that's a good change

**Review by mcovarr — APPROVED** [review]
👍

**Pull request merged**